server/src/main/java/org/apache/iotdb/db/engine/tsfilemanagement/TsFileManagement.java

```diff
@@ -19,24 +19,56 @@
 
 package org.apache.iotdb.db.engine.tsfilemanagement;
 
+import static org.apache.iotdb.db.engine.storagegroup.StorageGroupProcessor.MERGING_MODIFICATION_FILE_NAME;
+
+import java.io.File;
 import java.io.IOException;
 import java.util.Iterator;
 import java.util.List;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.engine.merge.manage.MergeManager;
+import org.apache.iotdb.db.engine.merge.manage.MergeResource;
+import org.apache.iotdb.db.engine.merge.selector.IMergeFileSelector;
+import org.apache.iotdb.db.engine.merge.selector.MaxFileMergeFileSelector;
+import org.apache.iotdb.db.engine.merge.selector.MaxSeriesMergeFileSelector;
+import org.apache.iotdb.db.engine.merge.selector.MergeFileStrategy;
+import org.apache.iotdb.db.engine.merge.task.MergeTask;
+import org.apache.iotdb.db.engine.modification.Modification;
+import org.apache.iotdb.db.engine.modification.ModificationFile;
 import org.apache.iotdb.db.engine.storagegroup.StorageGroupProcessor.CloseHotCompactionMergeCallBack;
 import org.apache.iotdb.db.engine.storagegroup.TsFileResource;
+import org.apache.iotdb.db.engine.tsfilemanagement.level.LevelTsFileManagement;
+import org.apache.iotdb.db.exception.MergeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public abstract class TsFileManagement {
 
+  private static final Logger logger = LoggerFactory.getLogger(TsFileManagement.class);
   protected String storageGroupName;
   protected String storageGroupDir;
+
+  /**
+   * mergeLock is to be used in the merge process. Concurrent queries, deletions and merges may
+   * result in losing some deletion in the merged new file, so a lock is necessary.
+   */
+  public ReentrantReadWriteLock mergeLock = new ReentrantReadWriteLock();
   /**
    * hotCompactionMergeLock is used to wait for TsFile list change in hot compaction merge
    * processor.
    */
   private final ReadWriteLock hotCompactionMergeLock = new ReentrantReadWriteLock();
 
+  public volatile boolean isUnseqMerging = false;
+  /**
+   * This is the modification file of the result of the current merge. Because the merged file may
+   * be invisible at this moment, without this, deletion/update during merge could be lost.
+   */
+  public ModificationFile mergingModification;
+  public long mergeStartTime;
```

Comment:
Please check the accessibility of **all new** fields ... Class variable fields should not have public accessibility
```suggestion
  private long mergeStartTime;
```

Reply:
accept it